### PR TITLE
ObjectLoader: make images use the loading manager specified in constructor

### PR DESCRIPTION
**Pull request by emeyex**
To address #13177 

Some example code using the new `imageManager` field:
```
const objLoader = new THREE.ObjectLoader( );
objLoader.imageManager = THREE.DefaultLoadingManager;
objLoader.load( url, onLoaded, onProgress, onError );
```

### CODE REVIEW

**Comment by Mugen87** [comment]
TBH, i would prefer a solution which is less "hacky" even if that means we are not fully backwards compatible. Such changes can be noted in the migration guide:

https://github.com/mrdoob/three.js/wiki/Migration-Guide 

So i vote to slightly refactor `ObjectLoader` in the way I mentioned in https://github.com/mrdoob/three.js/issues/13177#issuecomment-360781725

**Comment by emeyex** [comment]
Sounds cleaner to me as well, I'm just used to being overly cautious on backward-compatibility! 😄 

So to clarify the changes:

- All images in the object being loaded will use the manager passed to the constructor of the `ObjectLoader` (else `DefaultLoadingManager` if nothing passed)
- No ability to change that ☝️ behavior via an option
- Does it make sense to have image load errors trigger the `onError` callbacks (i.e., `parse` and `parseImages` would now also accept `onProgress, onError` and `load` would forward its arguments on)?

**Comment by emeyex** [comment]
Checked the behavior of `GLTFLoader`, it looks like image load errors do indeed trigger `onError` callbacks, so making the changes above would make things more consistent between `ObjectLoader` and `GLTFLoader`. Will push up changes matching the above bullet points.

File: src/loaders/ObjectLoader.js [hunk 1]
```diff
@@ -477,9 +477,13 @@ Object.assign( ObjectLoader.prototype, {
 
 			return loader.load( url, function () {
 
+				if ( onLoad !== undefined ) onLoad( );
```
Comment (Mugen87) [inline src/loaders/ObjectLoader.js]: I'm afraid this change is incorrect. The code will fire the global `onLoad()` function as soon as a single image is loaded. If we only use a single instance of `LoadingManager`, it's necessary to rewrite `parseImages()`.

Comment (Mugen87) [reply src/loaders/ObjectLoader.js]: I suggest to close the PR and wait on @mrdoob's opinion. We should continue the discussion in #13177 and open a new PR if it's clear what we're going do...

Comment (emeyex) [reply src/loaders/ObjectLoader.js]: Good catch, of course the loading manager isn't gating the callback anymore. I grabbed a model with multiple textures for testing and pushed the fix. Obviously it's up to you guys what PRs you accept, but IMO this is a fix for a non-trivial oversight in the core model loader.

**Comment by Mugen87** [comment]
Closing for now. If we actually change `ObjectLoader`, we need to implement that differently.

**Comment by emeyex** [comment]
I am curious to know how it needs to be implemented differently. Is there an issue with this PR as it currently stands? I believe it matches the guidelines you laid out in #13177, and passes my tests using several different objects (some with multiple textures). If there is a specific test suite I can run it through, I would also be more than happy to.

**Pull request closed without merging**